libraries/AP_HAL_Linux/SPIUARTDriver.cpp

```diff
@@ -122,7 +99,7 @@ int SPIUARTDriver::_write_fd(const uint8_t *buf, uint16_t size)
 
     _dev->get_semaphore()->give();
 
-    BUF_ADVANCEHEAD(_writebuf, size);
+    _writebuf.advance(size);
```

Comment:
yes, you are right... I think I focused on carefully reviewing the PX4 version and forgot the others.